### DISCUSSION TITLE
Drudg the e-VLBI vex files with the keyword target_correlator

### DISCUSSION THREAD

**Discussion by vlbi-jun-yang** [nvi-inc/fs#162] [Ideas]
Currently, stations have to manually change the recorder to none to _drudg_ e-VLBI vex files. If _recorder=none_ is defined in the vex files, station might let _drudg_ to overwrite the local setup and then _drudg_ e-VLBi file automatically. To investigate its feasibility, I have communicated with JIVE. This requires JIVE to provide another input station data file for users to run the NRAO _Sched_ or the JIVE _py-Sched_.  This may cause additional issues. So, in practice, it is not promising than I thought. 

To avoid the manual change of the recorder, I have wrote a local script to comment out the commands dedicated to the disk-recording observations.  Still, I need an option for my script to define the e-VLBI mode. Later, I propose JIVE to use the keyword _target_correlator_ to define the e-VLBI mode clearly in the vex files.  We have agreed on seting _target_correlator = JIVE_E-VLBI_. The keyword is not really used by _drudg_. So,  _drudg_  is happy for the new value.  

At another level, I am wondering whether we can get _drudg_ to use the keyword as an alternative way of _recorder=none_ for the e-VLBI observations.

**Comment by wehimwich** [comment]
Too bad about  `recorder=none` not being feasible.  If don't have `equipment_overide` enabled in _skedf.ctl_, then _drudg_ will respect the equipment in the schedule.

If you don't have optional prompts (`tpicd` etc) from _drudg_, your script could use _bash_ `here document` a la _fesh_:

```
cat <<EOF | drudg "$schedule"
oe
11
19 1 0 0
12
5
0
EOF
```
You would have to make sure there is no _.snp_ or _.prc_ file or _drudg_ will prompt to overwrite, which won't agree with the prepared responses.

Actually, if you have optional prompts, you could make that work too, if the prompts and answers are predictable.

**Reply by vlbi-jun-yang** [reply]
Thanks for the suggestion. Great to know the new way to run _drudg_.

**Reply by wehimwich** [reply]
You can even put variables in the `here document`. That can require some gymnastics for answers to questions that may or may not be asked. There are examples in _fesh_ (thanks @dehorsley), including handling optional questions.

**Reply by vlbi-jun-yang** [reply]
Thanks for the comments. Indeed, we can use variables. While, we still require an option to take care of the e-VLBI vex files in the command line. Station operators have to pay good attention to the option. Now, the keyword `target_correlator = JIVE_E-VLBI`  help me to remove the worry permanently.

**Reply by wehimwich** [reply]
I think it is better to not have multiple paths for information from the schedule file. We could add an optional keyword to _skedf.ctl_  so that if the string `e-vlbi ` (case insensitive) appears in `target_correlator`, _drudg_ will force the recorder to `none`.

I am not sure how e-VLBI will be work for the DBBC3. I am guessing that  would involve a change to the Ethernet configuration of the device. Perhaps, _drudg_ could not only set the recorder to `none`, but could also insert a call to station defined procedure that will change the ethernet configuration for the target correlator. This assumes we can change the Ethernet configuration without having to re-sync. I guess we would also need a non-e-VLBI ethernet configuration procedure. Maybe all that would have utility for the DBBC(2) as well.

**Reply by haavee** [reply]
Typically, during e-VLBI, it is the correlator control who changes (parts of) the FiLa10G or jive5ab network parameters to be able to route the traffic to the machine(s) that are available and in the form (size) of network packets that the route between the station and the correlator actually supports.

**Reply by wehimwich** [reply]
Thanks. I forgot about that. Presumably that will work for the DBBC3 as well. If so, _drudg_ only needs to set the recorder to none.

In retrospect, I think my concern about the DBBC3 and e-VLBI is that I think I heard that the correlator dynamically changes what data are being transmitted (number of channels? bits/channel? ...?). If so, that might be challenging with the current DBBC3 firmware, which I think requires a re-sync for such changes, but maybe I am misremembering.

**Reply by vlbi-jun-yang** [reply]
I like the idea of adding an optional keyword to skedf.ctl to force the recorder to `none` in case of `target_correlator = JIVE_E-VLBI `. Look forward to seeing the new feature.
